source/common/upstream/outlier_detection_impl.cc

```diff
@@ -94,16 +110,94 @@ Http::Code DetectorHostMonitorImpl::resultToHttpCode(Result result) {
   case Result::REQUEST_FAILED:
     http_code = Http::Code::InternalServerError;
     break;
-  case Result::SERVER_FAILURE:
-    http_code = Http::Code::ServiceUnavailable;
+  case Result::CONNECT_SUCCESS:
```

Comment:
It makes me a little uncomfortable that we are not handling an enum value in this way. Should the enums actually be split into local/external enums, and then how those are handled depends on whether splitting is on or not? WDYT?

Reply:
Processing of enums is actually split depending on the mode: (split vs non-split):
For non-split mode: https://github.com/envoyproxy/envoy/blob/4ba5f898e20f7f8e7a228c4ad78b81b43971a736/source/common/upstream/outlier_detection_impl.cc#L124-L125 and for split mode: https://github.com/envoyproxy/envoy/blob/4ba5f898e20f7f8e7a228c4ad78b81b43971a736/source/common/upstream/outlier_detection_impl.cc#L140-L141.

Also see my response to your comment titled: _Shouldn't this be CONNECT_SUCCESS?_